src/System.IO.Pipelines/src/System/IO/Pipelines/StreamExtensions.netstandard.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    // Helpers to write Memory<byte> to Stream on netstandard 2.0
+    internal static class StreamExtensions
+    {
+        public static ValueTask<int> ReadAsync(this Stream stream, Memory<byte> buffer, CancellationToken cancellationToken = default)
```

Comment:
Same argument validation comments.  I'll stop mentioning this; just assume it applies everywhere :)

Reply:
Why? This is an internal method. The Stream can't be null because the public methods check for that before calling into this.

Reply:
Ok, as long as someone is validating :)